ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -23,26 +25,44 @@ class RetentionEventsQuery(ClickhouseEventQuery):
 
     def __init__(self, event_query_type: RetentionQueryType, *args, **kwargs):
         self._event_query_type = event_query_type
-        super().__init__(*args, **kwargs)
+        super().__init__(
+            *args,
+            should_join_persons=True,
+            extra_person_fields=["person_props"] if event_query_type != RetentionQueryType.RETURNING else [],
+            **kwargs,
+        )
 
         self._trunc_func = get_trunc_func_ch(self._filter.period)
 
     def get_query(self) -> Tuple[str, Dict[str, Any]]:
         _fields = [
             self.get_timestamp_field(),
             f"{get_aggregation_target_field(self._filter.aggregation_group_type_index, self.EVENT_TABLE_ALIAS, self.DISTINCT_ID_TABLE_ALIAS)} as target",
-            (
-                f"argMin(e.uuid, {self._trunc_func}(e.timestamp)) as min_uuid"
-                if self._event_query_type == RetentionQueryType.TARGET_FIRST_TIME
-                else f"{self.EVENT_TABLE_ALIAS}.uuid AS uuid"
-            ),
-            (
-                f"argMin(e.event, {self._trunc_func}(e.timestamp)) as min_event"
-                if self._event_query_type == RetentionQueryType.TARGET_FIRST_TIME
-                else f"{self.EVENT_TABLE_ALIAS}.event AS event"
-            ),
```

Comment:
I should probably add a comment on this, and perhaps shouldn't have done it, although imo makes the additions for breakdown_values look a little less messy. 

The argMin was to get values from the first event when `TARGET_FIRST_TIME` is specified by looking for the value for minimum timestamp. Rather than do this per select, we can just use LIMIT 1 BY target, which means that the result set will be the [first row for each distinct target](https://clickhouse.com/docs/en/sql-reference/statements/select/limit-by/).

I'll add a performance benchmark label to check we get comparable performance.

Reply:
I've just reverted these changes and instead updated `get_single_or_multi_property_string_expr` to accommodate requirements, to avoid distractions 🙏